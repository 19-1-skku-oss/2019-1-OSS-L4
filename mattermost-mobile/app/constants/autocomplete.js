// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const AT_MENTION_REGEX = /\B(@([^@\r\n\s]*))$/i;

export const AT_MENTION_SEARCH_REGEX = /\bfrom:\s*(\S*)$/i;

export const CHANNEL_MENTION_REGEX = /\B(~([^~\r\n]*))$/i;

export const CHANNEL_MENTION_SEARCH_REGEX = /\b(?:in|channel):\s*(\S*)$/i;

export const DATE_MENTION_SEARCH_REGEX = /\b(?:on|before|after):\s*(\S*)$/i;

export const ALL_SEARCH_FLAGS_REGEX = /\b\w+:/g;