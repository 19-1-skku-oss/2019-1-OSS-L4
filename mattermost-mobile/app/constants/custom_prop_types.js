// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import PropTypes from 'prop-types';

export const Children = PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf([PropTypes.node])]);

export const Style = PropTypes.oneOfType([
    PropTypes.object, // inline style
    PropTypes.number, // style sheet entry
    PropTypes.array,
]);

export default {
    Children,
    Style,
};
