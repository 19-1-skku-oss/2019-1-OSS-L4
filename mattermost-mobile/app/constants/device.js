// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeviceInfo from 'react-native-device-info';
import RNFetchBlobFS from 'rn-fetch-blob/fs';
import keyMirror from 'mattermost-redux/utils/key_mirror';

const deviceTypes = keyMirror({
    CONNECTION_CHANGED: null,
    DEVICE_DIMENSIONS_CHANGED: null,
    DEVICE_TYPE_CHANGED: null,
    DEVICE_ORIENTATION_CHANGED: null,
    STATUSBAR_HEIGHT_CHANGED: null,
});

export default {
    ...deviceTypes,
    DOCUMENTS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Documents`,
    IMAGES_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Images`,
    IS_IPHONE_X: DeviceInfo.getModel().includes('iPhone X'),
    IS_TABLET: DeviceInfo.isTablet(),
    VIDEOS_PATH: `${RNFetchBlobFS.dirs.CacheDir}/Videos`,
};
