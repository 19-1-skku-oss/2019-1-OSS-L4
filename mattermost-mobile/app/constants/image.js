// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const IMAGE_MAX_HEIGHT = 350;
export const IMAGE_MIN_DIMENSION = 50;
