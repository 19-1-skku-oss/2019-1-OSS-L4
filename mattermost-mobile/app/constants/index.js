// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import PermissionTypes from './permissions';
import ViewTypes, {UpgradeTypes} from './view';

export {
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    PermissionTypes,
    UpgradeTypes,
    ViewTypes,
};
