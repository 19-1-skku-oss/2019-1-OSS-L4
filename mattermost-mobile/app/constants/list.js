// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const VISIBILITY_CONFIG_DEFAULTS = {
    viewAreaCoveragePercentThreshold: 3,
    waitForInteraction: false,
};

export default {
    VISIBILITY_CONFIG_DEFAULTS,
    VISIBILITY_SCROLL_DOWN: 'down',
    VISIBILITY_SCROLL_UP: 'up',
};
