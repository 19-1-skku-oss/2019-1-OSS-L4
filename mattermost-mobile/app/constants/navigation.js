// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

const NavigationTypes = keyMirror({
    NAVIGATION_RESET: null,
    NAVIGATION_CLOSE_MODAL: null,
    NAVIGATION_NO_TEAMS: null,
    RESTART_APP: null,
    NAVIGATION_ERROR_TEAMS: null,
});

export default NavigationTypes;
