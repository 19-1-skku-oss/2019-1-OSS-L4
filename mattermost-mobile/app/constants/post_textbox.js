// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_CONTENT_HEIGHT = 100;
export const MAX_FILE_COUNT = 5;
export const IS_REACTION_REGEX = /(^\+:([^:\s]*):)$/i;
export const INSERT_TO_DRAFT = 'insert_to_draft';
export const INSERT_TO_COMMENT = 'insert_to_comment';
