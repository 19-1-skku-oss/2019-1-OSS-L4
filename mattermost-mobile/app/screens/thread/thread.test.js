// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';
import {General, RequestStatus} from 'mattermost-redux/constants';
import PostList from 'app/components/post_list';

import Thread from './thread.js';

jest.mock('react-intl');

describe('thread', () => {
    const navigator = {
        dismissModal: jest.fn(),
        pop: jest.fn(),
        resetTo: jest.fn(),
        setTitle: jest.fn(),
    };
    const baseProps = {
        actions: {
            selectPost: jest.fn(),
        },
        channelId: 'channel_id',
        channelType: General.OPEN_CHANNEL,
        displayName: 'channel_display_name',
        navigator,
        myMember: {last_viewed_at: 0, user_id: 'member_user_id'},
        rootId: 'root_id',
        theme: Preferences.THEMES.default,
        postIds: ['root_id', 'post_id_1', 'post_id_2'],
        channelIsArchived: false,
        threadLoadingStatus: {status: RequestStatus.STARTED},
    };

    test('should match snapshot, has root post', () => {
        const wrapper = shallow(
            <Thread {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, no root post, loading', () => {
        const newPostIds = ['post_id_1', 'post_id_2'];
        const wrapper = shallow(
            <Thread
                {...baseProps}
                postIds={newPostIds}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call props.navigator on onCloseChannel', () => {
        const channelScreen = {
            screen: 'Channel',
            title: '',
            animated: false,
            backButtonTitle: '',
            navigatorStyle: {
                animated: true,
                animationType: 'fade',
                navBarHidden: true,
                statusBarHidden: false,
                statusBarHideWithNavBar: false,
                screenBackgroundColor: 'transparent',
            },
            passProps: {
                disableTermsModal: true,
            },
        };
        const newNavigator = {...navigator};
        const wrapper = shallow(
            <Thread
                {...baseProps}
                navigator={newNavigator}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        wrapper.instance().onCloseChannel();
        expect(newNavigator.resetTo).toHaveBeenCalledTimes(1);
        expect(newNavigator.resetTo).toBeCalledWith(channelScreen);
    });

    test('should match snapshot, render footer', () => {
        const wrapper = shallow(
            <Thread {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        // return loading
        expect(wrapper.find(PostList).getElement()).toMatchSnapshot();

        // return null
        wrapper.setProps({threadLoadingStatus: {status: RequestStatus.SUCCESS}});
        expect(wrapper.find(PostList).getElement()).toMatchSnapshot();

        // return deleted post
        const newPostIds = ['post_id_1', 'post_id_2'];
        wrapper.setProps({postIds: newPostIds});
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
