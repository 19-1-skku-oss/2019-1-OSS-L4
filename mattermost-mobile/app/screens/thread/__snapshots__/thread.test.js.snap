// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread should match snapshot, has root post 1`] = `
<React.Fragment>
  <Connect(SafeAreaIos)
    excludeHeader={true}
    keyboardOffset={20}
  >
    <Connect(StatusBar) />
    <React.Fragment>
      <Connect(PostList)
        currentUserId="member_user_id"
        indicateNewMessages={false}
        lastPostIndex={-1}
        location="thread"
        navigator={
          Object {
            "dismissModal": [MockFunction],
            "pop": [MockFunction],
            "resetTo": [MockFunction],
            "setTitle": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "title": undefined,
                  },
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
          }
        }
        onPostPress={[Function]}
        postIds={
          Array [
            "root_id",
            "post_id_1",
            "post_id_2",
          ]
        }
        renderFooter={
          <Loading
            color="grey"
            size="large"
            style={Object {}}
          />
        }
        scrollViewNativeID="threadPostList"
      />
      <Connect(FileUploadPreview)
        channelId="channel_id"
        rootId="root_id"
      />
      <ForwardRef(forwardConnectRef)
        cursorPositionEvent="onThreadTextBoxCursorChange"
        maxHeight={200}
        onChangeText={[Function]}
        rootId="root_id"
        valueEvent="onThreadTextBoxValueChange"
      />
    </React.Fragment>
  </Connect(SafeAreaIos)>
  <KeyboardTrackingView
    scrollViewNativeID="threadPostList"
  >
    <ForwardRef(forwardConnectRef)
      channelId="channel_id"
      channelIsArchived={false}
      cursorPositionEvent="onThreadTextBoxCursorChange"
      navigator={
        Object {
          "dismissModal": [MockFunction],
          "pop": [MockFunction],
          "resetTo": [MockFunction],
          "setTitle": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "title": undefined,
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      onCloseChannel={[Function]}
      rootId="root_id"
      valueEvent="onThreadTextBoxValueChange"
    />
  </KeyboardTrackingView>
</React.Fragment>
`;

exports[`thread should match snapshot, no root post, loading 1`] = `
<React.Fragment>
  <Connect(SafeAreaIos)
    excludeHeader={true}
    keyboardOffset={20}
  >
    <Connect(StatusBar) />
    <Loading
      color="grey"
      size="large"
      style={Object {}}
    />
  </Connect(SafeAreaIos)>
</React.Fragment>
`;

exports[`thread should match snapshot, render footer 1`] = `
<Connect(PostList)
  currentUserId="member_user_id"
  indicateNewMessages={false}
  lastPostIndex={-1}
  location="thread"
  navigator={
    Object {
      "dismissModal": [MockFunction],
      "pop": [MockFunction],
      "resetTo": [MockFunction],
      "setTitle": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "title": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  onPostPress={[Function]}
  postIds={
    Array [
      "root_id",
      "post_id_1",
      "post_id_2",
    ]
  }
  renderFooter={
    <Loading
      color="grey"
      size="large"
      style={Object {}}
    />
  }
  scrollViewNativeID="threadPostList"
/>
`;

exports[`thread should match snapshot, render footer 2`] = `
<Connect(PostList)
  currentUserId="member_user_id"
  indicateNewMessages={false}
  lastPostIndex={-1}
  lastViewedAt={0}
  location="thread"
  navigator={
    Object {
      "dismissModal": [MockFunction],
      "pop": [MockFunction],
      "resetTo": [MockFunction],
      "setTitle": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "title": undefined,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
  onPostPress={[Function]}
  postIds={
    Array [
      "root_id",
      "post_id_1",
      "post_id_2",
    ]
  }
  renderFooter={null}
  scrollViewNativeID="threadPostList"
/>
`;

exports[`thread should match snapshot, render footer 3`] = `
<React.Fragment>
  <Connect(SafeAreaIos)
    excludeHeader={true}
    keyboardOffset={20}
  >
    <Connect(StatusBar) />
    <Loading
      color="grey"
      size="large"
      style={Object {}}
    />
  </Connect(SafeAreaIos)>
</React.Fragment>
`;
